baseplate/server/__init__.py

```diff
@@ -190,6 +214,94 @@ def configure_logging(config: Configuration, debug: bool) -> None:
+    logger.info("Entering configure tracing function")
+    if config.tracing:
+        logger.info("Tracing config has been provided")
+        if "service_name" in config.tracing and "endpoint" in config.tracing:
+            resource = Resource.create({SERVICE_NAME: config.tracing["service_name"]})
+            # Don't sample unless parent sampled by default
+            sampler = DEFAULT_OFF
+            if "sample_rate" in config.tracing:
+                try:
+                    sample_rate = float(config.tracing["sample_rate"])
+                except ValueError:
+                    logger.error("unable to parse tracing sample_rate")
+                    return
+                if sample_rate is not None and 0.0 <= sample_rate <= 1.0:
+                    if sample_rate == 1.0:
+                        # if sample rate is 1.0 sample by default
+                        sampler = DEFAULT_ON
+                    elif sample_rate == 0.0:
+                        sampler = DEFAULT_OFF
+                    else:
+                        sampler = ParentBasedTraceIdRatio(sample_rate)
+                else:
+                    # if you don't set a sample rate we wont sample by default.
+                    logger.warning(
+                        "sample_rate invalid in tracing config, it must be a float between 0 and 1. Defaulting to no sampling."
+                    )
+            else:
+                logger.warning(
+                    "sample_rate not specified in tracing config, default to no sampling."
+                )
+            # Load insecure option from tracing config.
+            insecure: bool
+            if config.tracing.get("insecure") == "false":
+                insecure = False
+            else:
+                # Default to true. At the time of implementation our current tracing pipeline is not setup for TLS.
+                insecure = True
+
+            otlp_exporter = OTLPSpanExporter(endpoint=config.tracing["endpoint"], insecure=insecure)
+            propagate.set_global_textmap(
+                CompositePropagator([RedditB3HTTPFormat(), TraceContextTextMapPropagator()])
+            )
+            provider = TracerProvider(sampler=sampler, resource=resource)
+            provider.add_span_processor(
+                BaseplateBatchSpanProcessor(
+                    otlp_exporter,
+                    attributes={
+                        ResourceAttributes.SERVICE_NAME: config.tracing["service_name"],
+                    },
+                )
+            )
+            trace.set_tracer_provider(provider)
+            otel_tracing_autoinstrumentation()
+            logger.info("Tracing Configured")
+        else:
+            logger.warning(
+                "Tracing disabled: service_name and endpoint are required fields in the tracing config."
+            )
+
+
+def otel_tracing_autoinstrumentation() -> None:
+    PyramidInstrumentor().instrument()
+    RequestsInstrumentor().instrument()
```

Comment:
I need to look at moving these back.

This created an infinite loop somehow related to the import timing of the `ssl` lib and our gevent `patch_all()` call in `baseplate-serve`. 

Honestly I don't 100% know why yet, still digging.

Reply:
Or an alternative that is not this 😓